doc/guide/go_quickstart.hpp

```diff
@@ -0,0 +1,241 @@
+
+  // Now print the accuracy.
+  rows, _ := predictions.Dims()
+  var sum int = 0
+  for i := 0; i < rows; i++ {
+    if (predictions.At(i, 0) == test_labels.At(i, 0)) {
+      sum = sum + 1
+    }
+  }
+  fmt.Println(sum, " correct of ", rows, "(", (float64(sum) / float64(rows)) *
+              100, "%)")
+}
+@endcode
+
+We can see that we achieve reasonably good accuracy on the test dataset (80%+);
+if we use the full @c covertype.csv.gz, the accuracy should increase
+significantly (but training will take longer).
+
+It's easy to modify the code above to do more complex things, or to use
+different mlpack learners, or to interface with other machine learning toolkits.
+
+@section go_quickstart_whatelse What else does mlpack implement?
+
+The example above has only shown a little bit of the functionality of mlpack.
+Lots of other commands are available with different functionality.  A full list
+of each of these commands and full documentation can be found on the following
+page:
+
+ - <a href="https://www.mlpack.org/doc/mlpack-git/go_documentation.html">Go documentation</a>
+
+You can also use the GoDoc to explore the @c mlpack module and its
+functions; every function comes with comprehensive documentation.
+
+For more information on what mlpack does, see https://www.mlpack.org/.
+Next, let's go through another example for providing movie recommendations with
+mlpack.
+
+@section go_quickstart_movierecs Using mlpack for movie recommendations
+
+In this example, we'll train a collaborative filtering model using mlpack's
+<tt><a href="https://www.mlpack.org/doc/mlpack-git/go_documentation.html#cf">cf()</a></tt> method.  We'll train this on the MovieLens dataset from
+https://grouplens.org/datasets/movielens/, and then we'll use the model that we
+train to give recommendations.
+
+You can copy-paste this code directly into main.go to run it.
+
+@code{.go}
+package main
+
+import (
+  "github.com/frictionlessdata/tableschema-go/csv"
+  "github.com/mlpack.org/v1/mlpack"
+  "gonum.org/v1/gonum/mat"
+  "fmt"
+  "os"
+)
+func main() {
+
+  // Download dataset.
+  mlpack.DownloadFile("https://www.mlpack.org/datasets/ml-20m/ratings-only.csv.gz",
+                      "ratings-only.csv.gz")
+  mlpack.DownloadFile("https://www.mlpack.org/datasets/ml-20m/movies.csv.gz",
+                      "movies.csv.gz")
+
+  // Extract dataset.
+  mlpack.UnZip("ratings-only.csv.gz", "ratings-only.csv")
+  f1, _ := os.Open("ratings-only.csv")
+  defer f1.Close()
+  data := mlpack.NewReader(f1)
+  _, _ = data.ReadHeading()
+  ratings, _ := data.ReadAll()
+
+  mlpack.UnZip("movies.csv.gz", "movies.csv")
+  table, _ := csv.NewTable(csv.FromFile("movies.csv"), csv.LoadHeaders())
+  movies, _ := table.ReadColumn("title")
+
+  // Split the dataset using mlpack.
+  params := mlpack.PreprocessSplitOptions()
+  params.TestRatio = 0.1
+  params.Verbose = true
+  ratings_test, _, ratings_train, _ :=
+      mlpack.PreprocessSplit(ratings, params)
```

Comment:
This is probably short enough for one line. :)
```suggestion
  ratings_test, _, ratings_train, _ := mlpack.PreprocessSplit(ratings, params)
```

Reply:
Done.